src/graph.rs

```diff
@@ -87,9 +94,17 @@ impl<'tree> Graph<'tree> {
         DisplayGraph(self)
     }
 
-    pub fn display_json(&self) {
+    pub fn display_json(&self, path: Option<&Path>) -> Result<()> {
         let s = serde_json::to_string_pretty(self).unwrap();
-        print!("{}", s)
+        path.map_or(stdout().write_all(s.as_bytes()), |path| {
+            File::create(path)?.write_all(s.as_bytes())
+        })
+        .with_context(|| {
+            format!(
+                "Failed to write JSON to {}",
+                path.map_or("stdout", |path| path.to_str().unwrap_or("output path"))
+            )
+        })
```

Comment:
Can't say I'm in love with this, but trying to abstract over `File` and `Stdout` wasn't going well, and I couldn't figure out appropriate conversions either. Same goes for `&str` and `Cow<str>`, hence some of the faffing about on line 105.

Reply:
I think this looks really nice, personally.